src/utils/createOmnichannelMessage.ts

```diff
@@ -35,18 +35,30 @@ const createOmnichannelMessage = (message: IRawMessage | ChatMessageReceivedEven
             type: PersonType.Bot
         } as IPerson;
 
-        if (metadata && metadata.amsMetadata && metadata.amsReferences) {
+        if (metadata) {
             try {
-                const references = JSON.parse(metadata.amsReferences);
-                const data = JSON.parse(metadata.amsMetadata);
-                const {fileName} = data[0];
-
                 omnichannelMessage.fileMetadata = {} as IFileMetadata; // Backward compatibility
+
+                if (metadata.amsMetadata) {
+                    const data = JSON.parse(metadata.amsMetadata);
+                    const {fileName} = data[0];
+                    omnichannelMessage.fileMetadata.name = fileName;
+                    omnichannelMessage.fileMetadata.type = fileName.split('.').pop();

```

Comment:
are we still using extensions?

Reply:
It's the omnichannel message contract. It's not the `File` object itself. We're just keeping backward compatibility there only.